runner.py

```diff
@@ -74,16 +74,17 @@
 
 def parallel_process(processing_dir, fastq_files, reference_file, quality_threshold, task, evalue, dust, num_alignments,
                      soft_masking, perc_identity, mode, reads_overlap):
-    process_fastq_partial = partial(process_fastq,  reference=reference_file, output_dir=processing_dir,
-                                    quality_threshold=quality_threshold, task=task, evalue=evalue, dust=dust,
-                                    num_alignments=num_alignments, soft_masking=soft_masking,
-                                    perc_identity=perc_identity, mode=mode, reads_overlap=reads_overlap)
-    try:
-        with concurrent.futures.ProcessPoolExecutor() as executor:
-            executor.map(process_fastq_partial, fastq_files)
-    except:
-        raise Exception("Processing failed. This may be due to RAM overload."
-                        "To avoid this try running with --max_memory max_available_RAM_in_MB")
+    with concurrent.futures.ProcessPoolExecutor() as executor:
+        future_tasks = {executor.submit(process_fastq, fastq_file, reference=reference_file, output_dir=processing_dir,
```

Comment:
why is this a dictionary and not a list? looks like you are only using the dictionary keys anyway...

Reply:
this is the correct syntax...!